scripts/generate_version_matrix.py

```
@@ -0,0 +1,79 @@
+import subprocess
+import json
+import re
+
+def get_git_tag():
+    try:
+        result = subprocess.run(["git", "describe", "--tags", "--exact-match"], capture_output=True, text=True, check=True)
+        return result.stdout.strip()
+    except subprocess.CalledProcessError:
+        return None
+
+def extract_name_and_version_from_tag(tag):
+    match = re.match(r"(pixi-build-[a-zA-Z-]+)-v(\d+\.\d+\.\d+)", tag)
+    if match:
+        return match.group(1), match.group(2)
+    return None, None
```

This should raise an Exception, not return `None`

---

we could have a situation when there is no tag for this commit ( just a simple push / PR )

---

But then this function will not be called.

You always have `if git_tag` beforehand

---

ah! sorry I've missed about what function you are talking!